### PR TITLE
Various stability fixes discovered during testing

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

**Comment by jimmidyson** [comment]
@bparees @gabemontero Wouldn't mind if you could take a look, especially around the dependent plugin upgrades?

File: pom.xml [hunk 1]
```diff
@@ -113,38 +113,101 @@
...
+      <version>1.5</version>
     </dependency>
 
     <!-- testing -->
     <dependency>
       <groupId>org.jenkins-ci.plugins.pipeline-stage-view</groupId>
       <artifactId>pipeline-stage-view</artifactId>
-      <version>1.0</version>
+      <version>2.1</version>
+      <scope>test</scope>
```
Comment (bparees) [inline pom.xml]: i think we're going to need to work with @tdawson to get all these plugin packages updated before our image can consume this new version of the plugin.  :/

**Comment by tdawson** [comment]
Can someone put those plugin/rpm changes in a nice list.  having everything in layered json isn't the easiest to determine everything that was changed.

Also, jenkins 1.651.3 ... gonna be hard to do .... just letting you know, it will take longer than that plugins.

**Comment by jimmidyson** [comment]
Ah I thought @bparees told me I could safely upgrade the Jenkins version...

**Comment by bparees** [comment]
@jimmidyson our rhel image currently includes 1.651.2, is that sufficient?  (our centos image has 1.651.3)

**Comment by jimmidyson** [comment]
thanks for the tip - let me retest with 1.65.2 but I can't imagine any problems with that.

**Comment by bparees** [comment]
@tdawson sure thing, here's the full list being introduced by @jimmidyson.  I did not double check to see if any of our plugin rpm packages are already at these levels, however:

```
git 3.0.0
git-client 2.0.0
workflow-api 2.5
workflow-step-api 2.4
workflow-durable-task-step 2.5
workflow-cps-global-lib 2.4
workflow-job 2.8
workflow-cps 2.21
workflow-basic-steps 2.2
workflow-support 2.10
pipeline-rest-api 2.1
structs 1.5
pipeline-stage-view 2.1
```

**Comment by jimmidyson** [comment]
Thanks @bparees.

**Comment by jimmidyson** [comment]
Note that pipeline-stage-view is only upgraded in test scope here for me to visualize stages while developing plugin - I guess it makes sense to upgrade anyway, but that is not a requirement of this sync plugin.

**Comment by jimmidyson** [comment]
I've just scoped dependencies properly. Required dependencies are:

git 3.0.0
git-client 2.0.0
workflow-api 2.5
workflow-job 2.8
workflow-cps 2.21
pipeline-rest-api 2.1

Recommended upgrades for pipeline jobs (but not direct runtime dependencies of this plugin) are:

workflow-step-api 2.4
workflow-durable-task-step 2.5
workflow-cps-global-lib 2.4
workflow-basic-steps 2.2
workflow-support 2.10
structs 1.5
pipeline-stage-view 2.1

**Comment by jimmidyson** [comment]
I'm going to self-merge this once tests have passed so I can push some changes that require this PR to be merged.

LGTM

**Comment by bparees** [comment]
thanks @jimmidyson!

@tdawson i'd prefer to update everything @jimmidyson listed but since they aren't absolute requirements, feel free to manage your other priorities as needed.

**Comment by tdawson** [comment]
Just double checked and we'll have to update all of the ones listed.
I have time today so I'll get started.  I'll start with the ones that have to get updated first and we'll see how far I can get.

**Comment by tdawson** [comment]
One quick question, looks like several of the packages are already newer than you have listed.  Can I use the newer versions or would you like the specific ones you listed.  Here is the list

pipeline-rest-api 2.1 (2.2)
pipeline-stage-view 2.1 (2.2)
workflow-basic-steps 2.2 (2.3)
workflow-cps 2.21 (2.22)
workflow-step-api 2.4 (2.5)

**Comment by jimmidyson** [comment]
Damn those over-productive developers... Upgrading is fine.

**Comment by tdawson** [comment]
Done:  plugins updated to 

git 3.0.0
git-client 2.0.0
pipeline-rest-api 2.2
pipeline-stage-view 2.2
structs 1.5
workflow-api 2.5
workflow-basic-steps 2.3
workflow-cps 2.22
workflow-cps-global-lib 2.4
workflow-durable-task-step 2.5
workflow-job 2.8
workflow-step-api 2.5
workflow-support 2.10

puddles have been created, images have been built with new puddles.
Both have been pushed to testing areas.

I'll work on getting the jenkins-1 rpm updated.  That's going to be trickier to get it into puddles.  For now, proceed as if I can't update it.

**Pull request merged**